webapp/tests/karma/ts/modules/contacts/contacts-content.component.spec.ts

```diff
@@ -240,4 +294,196 @@ describe('Contacts content component', () => {
+      userSettingsService.get.resolves({ facility_id: 'district-123' });
+
+      component.ngOnInit();
+      flush();
+
+      expect(globalActions.setRightActionBar.callCount).to.equal(1);
+      expect(globalActions.setRightActionBar.args[0][0].canDelete).to.equal(false);
+      expect(globalActions.setRightActionBar.args[0][0].canEdit).to.equal(false);
+    }));
+
+    it('should enable edit when user is not admin and facility is not home place ', fakeAsync(() => {
+      sinon.resetHistory();
+      sessionService.isAdmin.returns(false);
+      component.userSettings = { facility_id: 'district-9' };
+
+      component.ngOnInit();
+      flush();
+
+      expect(globalActions.setRightActionBar.callCount).to.equal(1);
+      expect(globalActions.setRightActionBar.args[0][0].canDelete).to.equal(false);
+      expect(globalActions.setRightActionBar.args[0][0].canEdit).to.equal(true);
+    }));
+
+    it('should filter contact types to allowed ones from all contact forms', fakeAsync(() => {
+      sinon.resetHistory();
+      contactTypesService.getChildren.resolves([
+        {
+          id: 'type1',
+          create_form: 'form:contact:create:type1',
+        },
+        {
+          id: 'type2',
+          create_form: 'form:contact:create:type2',
+        },
+        {
+          id: 'type3',
+          create_form: 'form:contact:create:type3',
+        },
+      ]);
+      const forms = [
+        { _id: 'form:contact:create:type3' },
+        { _id: 'form:contact:create:type2' },
+      ];
+
+      component.ngOnInit();
+      flush();
+
+      expect(xmlFormsService.subscribe.callCount).to.equal(2);
+      expect(xmlFormsService.subscribe.args[0][0]).to.equal('ContactsReportsForms');
+      expect(xmlFormsService.subscribe.args[0][1]).to.deep.equal({ contactForms: true });
+
+      xmlFormsService.subscribe.args[0][2](null, forms);
+
+      expect(globalActions.updateRightActionBar.callCount).to.equal(1);
+      expect(globalActions.updateRightActionBar.args[0][0]).to.deep.equal({
+        childTypes: [
+          {
+            menu_icon: 'fa-building',
+            menu_key: 'Add place',
+            permission: 'can_create_places',
+            types: [
+              {
+                create_form: 'form:contact:create:type2',
+                id: 'type2'
+              },
+              {
+                create_form: 'form:contact:create:type3',
+                id: 'type3'
+              }
+            ]
+          }
+        ]
+      });
+    }));
+
+    it('should set relevant report forms based on the selected contact', fakeAsync(() => {
+      sinon.resetHistory();
+      contactTypesService.getChildren.resolves([
+        {
+          id: 'type1',
+          create_form: 'form:test_report:type1',
+        },
+        {
+          id: 'type2',
+          create_form: 'form:test_report:type2',
+        },
+        {
+          id: 'type3',
+          create_form: 'form:test_report:type3',
+        },
+      ]);
```

Comment:
I don't think these need to be here, children shouldn't matter in this test, as we only care about the selected contact. The fact that they have the same `create_form` as the report forms you list below is just confusing. 
It's like saying that a pregnancy form is the same form that creates a clinic.

Reply:
For tracking, addressed here: https://github.com/medic/cht-core/pull/6864